docs/NFF-0.1-spec.norg

```diff
@@ -0,0 +1,875 @@
+    ranged and non-ranged detached modifier, as both have the same meaning, one just allows more
+    content to exist within itself.
+
+*** Custom Detached Modifiers
+    Apart from linking to the detached modifiers outlined above, you can also link to a set of
+    custom modifiers specifically designed for links. These are the `#` (magic) and `@` (file)
+    linkable locations.
+
+**** The File Linkable (`@`)
+     Sometimes you may want to link to an external file that is not a _Norg_ file.
+     In that case you may use `{@ /path/to/my/file.txt}` (notice the mandatory space after the `@`
+     char, just like with the {**** detached modifier}s). Paths are relative to the _Norg_
+     file that contains the link.
+
+**** The Magic Char (`#`)
+     Sometimes you simply want to be lazy, or you want to link to an {**** inline linkables}[inline
+     linkable] that does not have a dedicated modifier to denote it - in these scenarios you would
+     use the magic char: `#`. It links to /any/ item type. The syntax is exactly the same as with
+     the other modifiers: `{# My Location}`.
+
+*** Inline Linkables
+    Although most linkable items are either {# Structural Detached Modifiers}[structural] or
+    {# Range-able Detached Modifiers}[range-able], there are also syntax elements in _Norg_ that
+    are inline - these are the `|name` {# Carryover Tags}[carryover tag] and the
+    {# Inline Link Target}[inline link target]. Both of these can be linked to through the
+    {# The Magic Char (`#`)}[the magic char].
+
+** Link Description
+   Link descriptions are denoted by square brackets: `[]`. They contain the description for either a
+   {# link location} (by placing the description after the link location), an {# anchors}[anchor]
+   definition (by placing the description before the link location) or an anchor declaration (where
+   the /only/ syntax item *is* the link description). Anchors are described {# anchors}[later on].
+
+** Links
+   Links in _Norg_ can exist as a standalone {# link location} in which case their text is used as
+   the link title (often makes sense for headings):
+
+   @code norg
+   {* I am a standalone link}
+   @end
+
+   When a custom description is required, it must be placed *_after_* the link location. This makes
+   sense in terms of writing as you first define where you link to, and then describe it further:
+
+   @code norg
+   {* I am a link}[with a description]
+   @end
+
+** Anchors
+   _Norg_ also has a concept called {# anchors}. These allow you to place a standalone
+   {# link description} inside of text (referred to as an <anchor declaration>).
+   The target which this anchor links to can then be _defined_ at another place in the document
+   with an <anchor definition> which is an initial {# link description} followed by a
+   {# link location}. This is especially useful when you want to link to the same target very
+   often, like for example a specific website.
+
```

Comment:
We need to talk about custom link descriptions in anchor definitions: `[anchor][my custom text]`